system/lib/wasmfs/js_api.cpp

```diff
@@ -246,6 +246,18 @@ int _wasmfs_close(int fd) {
   return __wasi_fd_close(fd);
 }
 
+int _wasmfs_mmap(size_t length, int prot, int flags, int fd, size_t offset) {
```

Comment:
```suggestion
int _wasmfs_mmap(size_t length, int prot, int flags, int fd, off_t offset) {
```
To match the type in `__syscall_mmap2`, which was changed with commit 21cc30d071ed17f2eddde21dec3a7c20df181d30.
https://github.com/emscripten-core/emscripten/blob/8b502b017e0f4cc677d5065ba378edebd283ec0d/system/lib/libc/musl/arch/emscripten/syscall_arch.h#L58

(However, I'm not sure if `{{{ defineI64Param('offset') }}}` is also needed in the JS layer)

Reply:
> (However, I'm not sure if `{{{ defineI64Param('offset') }}}` is also needed in the JS layer)

Ah, it needs to be defined with `{{{ splitI64('offset') }}}` during the invocation, just like `llseek`.
https://github.com/emscripten-core/emscripten/blob/8b502b017e0f4cc677d5065ba378edebd283ec0d/src/library_wasmfs.js#L371

FWIW, I updated the above suggestion to also use the correct return type.

Reply:
Sorry, I just saw your reply. I definitely agree with the use of `{{{ splitI64('offset') }}}`.

Reply:
Great! I think using only `{{{ splitI64('offset') }}}` is fine, it looks like `{{{ defineI64Param('offset') }}}` is used for native -> JS calls.